### ISSUE TITLE
Forward not working for AdvancederDrive, AdvancedTurn occasionally choppy

### ISSUE THREAD

**Issue by bugeybot** [FRCTeam1073-TheForceTeam/robot19_VSCode#22]
AD: When setting direction to "forward" it spins.

AT: Motors seem to cut out
AT: Sometimes overshoots

**Comment by BenWertz** [comment]
AD: Gyro problem?
Maybe AdvancederDrive accidentally, instead of turning in the direction that keeps the gyro heading constant, turns the opposite way, so instead of adjusting to keep going straight, turns the other way, and turns more and more the further off it gets.

**Comment by bugeybot** [comment]
I had the same thought but when the adjustment speed only limits by 15% it wouldn't look like a full spin if that was the issue. It may however be an issue to solve later so good catch.

**Comment by BenWertz** [comment]
Found this in AdvancederDrive.java:
`Robot.drivetrain.tank(distance * encoderCheck() * lidarCheck() * gyroCheck("left"), direction * encoderCheck() * lidarCheck() * gyroCheck("right"));`

It's setting
`distance*encoderCheck()*lidarCheck()*gyroCheck("left")`
on the left side, but
`direction*encoderCheck()*lidarCheck()*gyroCheck("right")`
on the right side.

It uses direction on the right, but distance on the left. Is that how it's supposed to work? I've looked in previous versions of the code, and it seems like they used to both be direction. Could that be it?
